tests/integration/test_tls.py

```diff
@@ -166,6 +166,11 @@ async def test_mattermost_db(ops_test: OpsTest) -> None:
             with attempt:
                 await check_tls_rewind(ops_test)
 
+        # Await for postgresql to be stable if not already
+        await ops_test.model.wait_for_idle(
+            apps=[DATABASE_APP_NAME], status="active", idle_period=15
+        )
+
```

Comment:
Is the charm not being completely ready the reason the test fails? If so, we should fix it differently.

We should figure out if it is related to the connection string being shared already in the relation data, but the database is not ready yet, or if it's for another reason that the failure happens.

This can be addressed in a separate PR.

Reply:
The motivation for this was to avoid errors like this one: https://github.com/canonical/postgresql-k8s-operator/actions/runs/9813699722/job/27100493808#step:24:652 where the charm fails to initialize the `db` relation. 

AFAICT this happens because we did not allow enough time for the charm to recover after running `await run_command_on_unit(ops_test, primary, "/charm/bin/pebble start postgresql")` so thats why I added this wait right after. 

Is the charm expected to recover immediately after this sort of restart (running `pebble stop/start` directly)? If so, then I agree, this is a bug and we should probably address the underlying issue in another PR. If not, then IMO its okay to keep this wait. Up to you guys

Reply:
Oh, actually I think I got your point: Even if the charm is not immediately ready after the restart, adding the relation should not fail, but instead delay the sharing of the connection string. Is this what you mean?

Reply:
Exactly.

However, after reading your comment about `pebble start` and thinking better, I think it's ok to leave it as is for now because we are doing some operations manually in the test (like the `pebble start`).

It would be good if the charm detects this situation (that the pebble service is offline) and removes the instance from the list of ready units (in the services used for the connection strings), but it would require more work. You can create a ticket so we can remember about that. Thanks!

Reply:
Created https://warthogs.atlassian.net/browse/DPE-4866 for future investigate this question. For the time being, I'll keep this then, to avoid CI suffering.